### PR TITLE
Adapt clients to Y2Users

### PR DESCRIPTION
**Pull request by dgdavid**
## Problem

`Y2Users` API was modified and extended. Client for creating users needs to be adapted.

* https://github.com/yast/yast-users/pull/329
* https://github.com/yast/yast-users/pull/332

## Solution

Adapt client accordingly.

NOTE: do not merge this branch into master until *yast-users/y2users* branch is merged.

NOTE2: test will pass once *yast-users/y2users* branch is merged into *master*.

## Testing

* Add unit tests.
* Manually tested.

### CODE REVIEW

File: src/lib/y2firstboot/clients/user.rb [hunk 1]
```diff
@@ -77,18 +80,21 @@ def run
...
+        user.home.path = home_path.dirname.join(user.name).to_s
       end
 
       # Writes config to the system
       def write_config
         writer = Y2Users::Linux::Writer.new(
           config,
-          Y2Users::ConfigManager.instance.system
+          Y2Users::ConfigManager.instance.system,
+          Y2Users::CommitConfigCollection.new.tap { |collection| collection.add(commit_config) }
```
Comment (joseivanlopez) [inline src/lib/y2firstboot/clients/user.rb:97]: NP: for one-liner blocks we use single letter for the params: ` { |c| c.add }`

Comment (joseivanlopez) [reply src/lib/y2firstboot/clients/user.rb:97]: BTW: I would extract the generation of the `CommitConfigCollection` to a separate method:

~~~
def commit_configs
  CommitConfigCollection.new.tap do |collection|
    config = CommitConfig.new.tag { ... }
    collection.add(config)
  end
end
~~~

**Review by joseivanlopez — APPROVED** [review]
Only NPs. BTW, I would not merge until testing it manually.

File: src/lib/y2firstboot/clients/user.rb [hunk 2]
```diff
@@ -156,6 +147,22 @@ def root_user
...
+      # Builds the commit configs to use when writing users
+      #
+      # @return [Y2Users::CommitConfigCollection]
+      def commit_configs
+        Y2Users::CommitConfigCollection.new.tap do |collection|
+          commit_config = Y2Users::CommitConfig.new.tap do |config|
+            config.username = user.name
+            config.move_home = true
+            config.adapt_home_ownership = true
+            config.home_without_skel = false
```
Comment (jreidinger) [inline src/lib/y2firstboot/clients/user.rb:159]: is somewhere reasoning why this values?

Comment (joseivanlopez) [reply src/lib/y2firstboot/clients/user.rb:159]: I will add a comment.

**Review by ancorgs — APPROVED** [review]
LGTM

**Pull request merged**